### PR TITLE
add session token to rulest

### PR DESCRIPTION
**Pull request by eze9252**
Hi team, this resolve:

- https://github.com/wazuh/wazuh-kibana-app/issues/3249

### CODE REVIEW

File: public/directives/wz-logtest/components/logtest.tsx [hunk 1]
```diff
@@ -81,7 +81,19 @@ export const Logtest = compose(
...
 
+  const getTokenSessionLogtest = async () => {
+    const body = {
+      log_format: 'syslog',
+      location: 'logtest',
+      event: 'get token logtest session',
+    };
+    return await WzRequest.apiReq('PUT', '/logtest', body)
+    .then(res => res.data.data.token)
+    .catch(error => '')
```
Comment (gabiwassan) [inline public/directives/wz-logtest/components/logtest.tsx:92]: Maybe we can return some value for `token`? 
`.catch(error => return '')`

**Review by matiasmoreno876 — APPROVED** [review]
FIXED

image


image

**Review by gabiwassan — APPROVED** [review]
LGTM !

**Review by pablomarga — APPROVED** [review]
LGTM!

File: public/directives/wz-logtest/components/logtest.tsx (hunk 1 above)
Comment (Desvelao) [inline public/directives/wz-logtest/components/logtest.tsx:92]: This code suggests to me that it is doing a request with a fake log to get the token. This should avoid.

Could we get the token with the request with the first log instead, and use it in the next request if there are more logs?

File: public/directives/wz-logtest/components/logtest.tsx [hunk 2]
```diff
@@ -91,6 +103,7 @@ export const Logtest = compose(
             log_format: 'syslog',
             location: 'logtest',
```
Comment (Desvelao) [inline public/directives/wz-logtest/components/logtest.tsx:104]: Maybe it is not related to the issue, but all the logs have these properties. The user can't try the Ruleset test utility with another type of log. If we give that possibility to select the log type, we should refactor the input log, replacing the text area for a form, with the log and type (i am not sure if we should include location too).

More info about another type of logs: https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/localfile.html#log-format

**Comment by Desvelao** [comment]
Some questions about the token:
1. Should we remove the session token after finish all requests? Using the Wazuh API request:
```
DELETE /logtest/sessions/{token}
```
https://documentation.wazuh.com/4.1/user-manual/api/reference.html#operation/api.controllers.logtest_controller.run_logtest_tool

2. Should we keep the token saved? If not, each time the user uses the Ruleset test, one token is generated.

File: public/directives/wz-logtest/components/logtest.tsx (hunk 1 above)
Comment (gabiwassan) [reply public/directives/wz-logtest/components/logtest.tsx]: Exactly! 👍🏼

**Comment by matiasmoreno876** [comment]
LGTM
image

**Review by matiasmoreno876 — APPROVED** [review]
LGTM

**Review by pablomarga — APPROVED** [review]
CR: :heavy_check_mark: 
Testing: :heavy_check_mark:

**Comment by Desvelao** [comment]
### Changes
- Use the token got in the first request in the next request

**Comment by Desvelao** [comment]
Should we close the log test session using the Wazuh API?

**Comment by CPAlejandro** [comment]
We proceed to add the following:
- [ ] The session token will be saved in Redux.
- [ ] A button to be able to close the session used.
- [ ] A modal to ensure that the button is not pressed to close the session by mistake.

**Review by mpRegalado — APPROVED** [review]
CR: LGTM! :heavy_check_mark: 
Testing: LGTM! :heavy_check_mark: 
RuleTest

File: public/directives/wz-logtest/components/logtest.tsx [hunk 3]
```diff
@@ -124,18 +147,41 @@ export const Logtest = compose(
...
+              deleteToken();
+            }}
+            color="danger"
+            modalTitle={`Do you want to clear current session?`}
+            modalProps={{
+              buttonColor: 'danger',
+              children: 'Clearing the session means the logs execution history is removed. This affects to rules that fire an alert when similar logs are executed in a specific range of time.'
+            }}
+          > 
+          Clear sesion
```
Comment (pablomarga) [inline public/directives/wz-logtest/components/logtest.tsx:181]: Typeerror it should be `session` instead of sesion

**Comment by pablomarga** [comment]
Testing: :heavy_check_mark: 
CR: check spell error

File: public/directives/wz-logtest/components/logtest.tsx (hunk 3 above)
Comment (CPAlejandro) [reply public/directives/wz-logtest/components/logtest.tsx]: I have already modified it, thanks you!

**Pull request merged**